src/Polly.Core/Utils/PipelineComponent.Composite.cs

```diff
@@ -0,0 +1,142 @@
+﻿using Polly.Telemetry;
+
+namespace Polly.Utils;
+
+internal abstract partial class PipelineComponent
+{
+    /// <summary>
+    /// A combination of multiple resilience strategies.
+    /// </summary>
+    [DebuggerDisplay("Pipeline, Strategies = {Strategies.Count}")]
+    [DebuggerTypeProxy(typeof(CompositeDebuggerProxy))]
+    internal sealed class CompositeComponent : PipelineComponent
+    {
+        private readonly PipelineComponent _firstComponent;
+        private readonly ResilienceStrategyTelemetry _telemetry;
+        private readonly TimeProvider _timeProvider;
+
+        private CompositeComponent(
+            PipelineComponent first,
+            IReadOnlyList<PipelineComponent> components,
+            ResilienceStrategyTelemetry telemetry,
+            TimeProvider timeProvider)
+        {
+            Components = components;
+
+            _telemetry = telemetry;
+            _timeProvider = timeProvider;
+            _firstComponent = first;
+        }
+
+        public static PipelineComponent Create(
+            IReadOnlyList<PipelineComponent> components,
+            ResilienceStrategyTelemetry telemetry,
+            TimeProvider timeProvider)
+        {
+            if (components.Count == 0)
+            {
+                throw new InvalidOperationException("The composite resilience pipeline must contain at least one resilience strategy.");
+            }
+
+            if (components.Distinct().Count() != components.Count)
+            {
+                throw new InvalidOperationException("The composite resilience pipeline must contain unique resilience strategies.");
+            }
```

Comment:
Should these exception messages be reworded to talk about components?

Reply:
Not quite sure here. Component is only internal detail while strategy is in the public API. I think both approaches are fine. What's your preference?

Reply:
Then maybe this is the wrong place to validate this? Feels weird to me for these exception messages to make assumptions about a higher level of the code?

Reply:
True, let me move this to pipeline builder.